designs/virtualization/4.5/Scheduling/Node-selector/node-selector.md

```diff
@@ -0,0 +1,37 @@
+---
+parent: Virtualization
+version: 4.5
+---
+
+# Node Selector
+
+Introducing Node-selector as it appears in-page and the Edit modal.
+
+![Node selector field in the Scheduling section ](img/Node-selector-0-0.jpg)
+The node-selector field is pulled from the Workload details section and placed under the Scheduling section.
+
+In this example there is no node-selector set, meaning, no node labels added.
+Clicking on the edit (pencil) button will open the Edit modal.
+Clicking on the "O labels" link text will open that modal as well.
+
+
+![Node selector modal ](img/Node-selector-1-0.jpg)
```

Comment:
I'm still unsure about using inline PatternFly alert styling for these waiting & checking states that aren't actually alerts. The grey color here would also need to be custom CSS, and the in-progress icon for this "waiting" state doesn't seem typical.

We definitely should signal to the user that this kind of validation will happen, but hopefully we can do so without using an inline alert. Maybe something like a dedicated "Matching nodes" section that just uses plain text until we reach the success/warning inline alert states:

![matching-nodes](https://user-images.githubusercontent.com/9122899/77130602-f59d7b80-6a2e-11ea-9200-db819b3acba1.png)

Reply:
I made, the changes u suggested. I think this not as good as having the entire Nodes checker in a box. Giving the user some visual anchor and helping her understand that some feedback is here. 
But I think we can move forward with this implementation fo now for the sake of the rest of the PRs depending on this one.

WDYT?

Reply:
I'm not 100% sure we can't use an inline info alert here. It feels like something we want to notify the user of and bring their attention to. I admit though this is somewhat uncharted territory in terms of inline alerts.
@Ranelim correct me if I'm wrong but this text is somewhat inaccurate
 "Please add at least one key/value pair above to check for matching nodes" 
I don't believe you have to add both key/value, you could just add just key without the value right?
If so we should tweak this text a bit to cover that.

Reply:
@matthewcarleton 


> @Ranelim correct me if I'm wrong but this text is somewhat inaccurate
> "Please add at least one key/value pair above to check for matching nodes"
> I don't believe you have to add both key/value, you could just add just key without the value right?

You are right. It went under my radar. This topic came up in our last Stakeholder meeting:
This is not actually a label. It could be a full label and it could be just a (full) Key and "Exist/nonExist" value. So we need to make modifications in the fields to support and surface that, as well as aligning this text.

For now, I want to share an alternative design to this Nodes Checker that uses "Info" notification instead of any custom design:

![Node-Checker-03](https://user-images.githubusercontent.com/48942365/77430014-2ee53b00-6de3-11ea-9d0a-dca0dc39d28d.jpg)

@matthewcarleton @andybraren 
WDYT?

Reply:
I think using an info alert here rather than a generic bell icon might be more appropriate. It might also be nice to add the spinner in the content area of the alert. WDYT? If you stick with the bell here I'd suggest using the pf colour rather than the info colour for the generic alert.

Reply:
> I think this not as good as having the entire Nodes checker in a box. Giving the user some visual anchor and helping her understand that some feedback is here.

In general I think the [usage and behavior guidelines](https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/alerts-and-notifications) should determine which component is used and ultimately how things look, rather than the other direction. I totally see the appeal of making this area look like one "widget" in a box though. See below.

> I'm not 100% sure we can't use an inline info alert here. It feels like something we want to notify the user of and bring their attention to. I admit though this is somewhat uncharted territory in terms of inline alerts.

Maybe my concern comes down to the wording/framing of the waiting and checking states.

As-is, I'm not sure "Waiting for input to check matching nodes" aligns with the [guidelines](https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/alerts-and-notifications)' suggestion to show alerts "as the result of a user action within a page." The "checking" state at least depends on user input/action. The wording in those guidelines admittedly has some wiggle room though. For some reason they only recommend placing inline alerts at the top of the page/content, but that's hardly followed. 🤷‍♂️

Maybe making the text a bit more "informational" and explanatory would help here. WIP, but something like:

Waiting state (info alert styling with the info icon):
> **Matching nodes will be searched for**
> Please add at least one label key above to search for nodes that match this selector.

Checking state (possibly the default alert styling with the bell, not sure):
> **Searching for matching nodes...**
> Please wait while nodes that match this selector are found.

Reply:
ya I agree we are pushing on alerts here a bit, but I agree it's not 100% clear and we can probably wiggle a bit here :) 
This could be a case for a unique component  b/c it has multiple states. I'm sure we could see this used in other places and having a recognizable pattern for dynamic results could be very nice. 
I think those suggestions work. I'm curious too how long we are talking for results to show. It could be almost instant, in that case we maybe don't see the searching state at all.

Reply:
> As-is, I'm not sure "Waiting for input to check matching nodes" aligns with the guidelines' suggestion to show alerts "as the result of a user action within a page."

I believe that infobox resolves that issue.

Adding two more options:
I think that if we are having a hard time with the Info notification box switching to alert, then the second option of support text is good as well. In fact, I'm starting to think this is better because it is not a notification. It pushes the right "volume" of attention and is not in par with the actual results.
As andy suggested, the intention was to create this "Smart widget box". But since it is perceived as a Notification and we cannot create something unique there is no point in mimicking an info notification pattern.

@andybraren @matthewcarleton  
WDYT

![Node-Checker-01](https://user-images.githubusercontent.com/48942365/77513159-2a6d6080-6e7d-11ea-82b3-0e8902e78c89.jpg)
![Node-Checker-02](https://user-images.githubusercontent.com/48942365/77513172-2e00e780-6e7d-11ea-8bcd-1db2b5fb87b8.jpg)

Reply:
@andybraren @matthewcarleton 
WDYT?
![Node-Checker-03](https://user-images.githubusercontent.com/48942365/77552816-af2b9f00-6ebc-11ea-9b14-91462fd3a26e.jpg)

Reply:
I like this! Seems like a nice combination of all concerns.

Reply:
Agreed, thank you for tolerating me. 😄

The descriptive text might also be a good opportunity to explain what the Node Selector is/does. Are "node labels or node keys" referring to the same thing here?

Reply:
> The descriptive text might also be a good opportunity to explain what the Node Selector is/does. Are "node labels or node keys" referring to the same thing here?

Yes, the user may enter entire label or just the Key of that label (to tap all the nodes with that Key)